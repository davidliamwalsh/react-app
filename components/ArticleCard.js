import { Component } from 'react'
import Link from 'next/link'
import moment from 'moment'

class ArticleCard extends Component {
  truncate (string) {
    if (string.length > 99) {
      return `${string.substr(0, 99)} ...`
    } else {
      return string
    }
  }

  render () {
    const { article } = this.props

    return <li className="c-articles__list">
      <div className="c-articles__list--hero">{article.image}</div>
      <div className="c-article__list__inner">
        <h4 className="c-articles__list__inner--title">{article.title}</h4>
        <p className="c-articles__list__inner--body">{this.truncate(article.body)}</p>
        <p className="c-articles__list__inner--date"><time dateTime={article.created_at}>{moment(article.created_at).format('LL')}</time><br />
          <Link href={`/post?id=${article.id}`}>
            <a>view more</a>
          </Link>
        </p>
      </div>
    </li>
  }
}

export default ArticleCard