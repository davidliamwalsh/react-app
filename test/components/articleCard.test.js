import ArticleCard from '../../components/ArticleCard'

describe('<ArticleCard />', () => {
  it('it renders correctly', () => {
    const article = {
      id: '1',
      title: 'This is my article',
      body: 'This is the body of my article',
      created_at: new Date(Date.UTC(96, 1, 2, 3, 4, 5))
    }

    const component = shallow(
      <ArticleCard article={article} />
    )

    expect(component).toMatchSnapshot()
  })
}) 