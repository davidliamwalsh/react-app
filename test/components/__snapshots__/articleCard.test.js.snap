// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> it renders correctly 1`] = `ShallowWrapper {}`;